archive/finder.go

```diff
@@ -46,21 +46,21 @@ func AddPath(pathField string, absolutePath bool) FindOption {
 			{"first", zng.TypeTime},
 			{"last", zng.TypeTime},
 		}
-		opt.addPath = func(ark *Archive, si SpanInfo, rec *zng.Record) (*zng.Record, error) {
+		opt.addPath = func(ark *Archive, chunk Chunk, rec *zng.Record) (*zng.Record, error) {
 			var path string
 			if absolutePath {
 				if ark.DataPath.Scheme == "file" {
-					path = si.LogID.Path(ark).Filepath()
+					path = chunk.Path(ark).Filepath()
```

Comment:
Did you consider adding an `ark *Archive` field to Chunk? Then you wouldn't have to pass ark to the Chunk methods.